sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/typedaggregators.scala

```diff
@@ -99,3 +97,67 @@ class TypedAverage[IN](val f: IN => Double) extends Aggregator[IN, (Double, Long
     toColumn.asInstanceOf[TypedColumn[IN, java.lang.Double]]
   }
 }
+
+class TypedMinDouble[IN](val f: IN => Double) extends Aggregator[IN, Double, Double] {
+  override def zero: Double = Double.PositiveInfinity
```

Comment:
Ah  I see my misunderstanding: in reduce I tried to also have an `if` for `f(a) == null` because of the previously mentioned implicit casting issue. This would force a `java.lang.Double` to be returned by the function, as `Double == null` doesn't make sense in Scala.

I have updated the code, please have a look :) Becuase `OUT` is already a `java.lang.Double`, we do not need the `toColumnJava`. As a result of `OUT` being `java.lang.Double` however, we do need a `toColumnScala` to accommodate `    val f = (x: (Double, Double)) => x._2; empty.agg(typed.min(f)).show()`